tests/integration/logging/monolog2/test_monolog_basic.php

```diff
@@ -0,0 +1,198 @@
+          "span.id": "??",
+          "entity.guid": "??",
+          "entity.name": "tests/integration/logging/monolog2__FILE__",
+          "hostname": "__HOST__"
+        },
+        {
+          "message": "notice",
+          "level": "NOTICE",
+          "timestamp": "??",
+          "trace.id": "??",
+          "span.id": "??",
+          "entity.guid": "??",
+          "entity.name": "tests/integration/logging/monolog2__FILE__",
+          "hostname": "__HOST__"
+        },
+        {
+          "message": "warning",
+          "level": "WARNING",
+          "timestamp": "??",
+          "trace.id": "??",
+          "span.id": "??",
+          "entity.guid": "??",
+          "entity.name": "tests/integration/logging/monolog2__FILE__",
+          "hostname": "__HOST__"
+        },
+        {
+          "message": "info",
+          "level": "INFO",
+          "timestamp": "??",
+          "trace.id": "??",
+          "span.id": "??",
+          "entity.guid": "??",
+          "entity.name": "tests/integration/logging/monolog2__FILE__",
+          "hostname": "__HOST__"
+        },
+        {
+          "message": "alert",
+          "level": "ALERT",
+          "timestamp": "??",
+          "trace.id": "??",
+          "span.id": "??",
+          "entity.guid": "??",
+          "entity.name": "tests/integration/logging/monolog2__FILE__",
+          "hostname": "__HOST__"
+        },  
+        {
+          "message": "emergency",
+          "level": "EMERGENCY",
+          "timestamp": "??",
+          "trace.id": "??",
+          "span.id": "??",
+          "entity.guid": "??",
+          "entity.name": "tests/integration/logging/monolog2__FILE__",
+          "hostname": "__HOST__"
+        },        
+        {
+          "message": "debug",
+          "level": "DEBUG",
+          "timestamp": "??",
+          "trace.id": "??",
+          "span.id": "??",
+          "entity.guid": "??",
+          "entity.name": "tests/integration/logging/monolog2__FILE__",
+          "hostname": "__HOST__"
+        }
+      ]
+    }
+  ]
+ */
+
+require_once(realpath(dirname(__FILE__)) . '/../../../include/config.php');
+require_once(realpath(dirname(__FILE__)) . '/../../../include/monolog.php');
+require_monolog(2);
+
+use Monolog\Logger;
+use Monolog\Handler\StreamHandler;
+use Monolog\Formatter\LineFormatter;
+
+
+function test_logging() {
+    $logger = new Logger('monolog2');
+
+    $logfmt = "%channel%.%level_name%: %message% %context%\n";
+    $formatter = new LineFormatter($logfmt);
+
+    $stdoutHandler = new StreamHandler('php://stdout', Logger::DEBUG);
+    $stdoutHandler->setFormatter($formatter);
+
+    $logger->pushHandler($stdoutHandler);
+    
+    // insert delays between log messages to allow priority sampling
+    // to resolve that later messages have higher precedence
+    // since timestamps are only millisecond resolution
+    // without delays sometimes order in output will reflect
+    // all having the same timestamp.
```

Comment:
Does the integration_runner care about the order of elements in EXPECT_LOG_EVENTS? If so, is the delay between events enough to guarantee that they will be sent by daemon in this order?

Reply:
Currently the runner does care.  If I did not give enough time between events that they had unique timestamps (log messages only stored with millisecond precision) the events would occasionally come out ordered differently.  This is because the tie breaker for event with the same priority is to favor newer events.  Since all had the same age normally this worked.  But sometimes the test ran and one (or more) events would have a different timestamp (ran the test right as millisecond flipped over to a new one) and it seemed to cause a different ordering.

For this test we don't really care about the order - just that the log events listed actually are created.  But the runner doesn't have a way to specify this currently - it does a exact match.  So this is probably the best we can do at the moment.  I don't believe as written it would ever produce a false positive which is good.